tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb

```diff
@@ -0,0 +1,801 @@
+    "                        'new_logon_security_id',\n",
+    "                        'network_information_source_network_address',\n",
+    "                        'detailed_authentication_information_transited_services',\n",
+    "                        'new_logon_account_domain',\n",
+    "                        'subject_account_domain',\n",
+    "                        'detailed_authentication_information_logon_process',\n",
+    "                        'account_for_which_logon_failed_account_domain',\n",
+    "                        'account_for_which_logon_failed_account_name',\n",
+    "                        'network_information_workstation_name',\n",
+    "                        'network_information_source_port',\n",
+    "                        'subject_security_id']\n",
+    "OUTPUT_DTYPES = ['str' for x in OUTPUT_COLS_SUPERSET]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parsing Data Pipeline\n",
+    "\n",
+    "Preprocess, or clean the raw logs, by removing the non-printable characters then begin to parse logs by event code type. Here we use regex mappings per event code to parse out each data element defined in OUTPUT_COLS_SUPERSET"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pipeline(df, event_codes, regex_mappings, clean=True):\n",
+    "    \"\"\"\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    if clean:\n",
+    "        df = preprocess_logs(df)\n",
+    "    out_dfs = [] \n",
+    "    \n",
+    "    # separate by eventcode and process differently\n",
+    "    \n",
+    "    for code in event_codes:\n",
+    "        portion = filter_by_pattern(df, code)\n",
+    "        temp = process_logtype(portion, regex_mappings, code)\n",
+    "        temp['time'] = portion['Time'].astype('int')\n",
+    "        temp['eventcode'] = portion['EventID']\n",
+    "        out_dfs.append(temp)\n",
+    "        \n",
+    "    # recombine the processed output\n",
+    "    out_df = cudf.multi.concat(out_dfs)\n",
+    "    return out_df\n",
+    "\n",
+    "\n",
+    "def concat_wrapper(df_list):\n",
+    "    return cudf.multi.concat(df_list)\n",
+    "\n",
+    "\n",
+    "def preprocess_logs(logs_gdf):\n",
+    "    \"\"\"Lowercasing and replacing characters\n",
+    "    \"\"\"\n",
+    "    logs_gdf['Raw'] = (\n",
+    "        logs_gdf['Raw'].str.lower()\n",
+    "        .str.replace('\\\\\\\\t', '')\n",
+    "        .str.replace('\\\\\\\\r', '')\n",
+    "        .str.replace('\\\\\\\\n', ' | ')\n",
+    "    )\n",
+    "    return logs_gdf\n",
+    "\n",
+    "\n",
+    "def process_logtype(df, regexes, eventcode):\n",
+    "    \"\"\"Ongoing strings development/fixes will allow for cleaner log processing code in the future\n",
+    "    \"\"\"\n",
+    "\n",
+    "    log_df_processed = cudf.read_csv('conf/LANL_OUTPUT_COLS_SUPERSET.csv', dtype=OUTPUT_DTYPES)\n",
```

Comment:
This currently fails due to https://github.com/rapidsai/cudf/issues/1364.

Since this is only being used to read the headers and decide the column names for the Dataframe (the csv itself has no values) one workaround could be to read it on host and then create the `log_df_processed` columns using these strings.

Reply:
Additional note: If you use a workaround, you could leave your original code commented and also add a comment above it with a link to the issue that blocks this. If the issue is resolved, users can go back to using your initial method rather than the workaround.

Reply:
I have made changes and included a link to the issue